lamps-rfc7030-csrattrs.mkd

```diff
@@ -153,6 +153,24 @@ MUST NOT include more than one element with a particiular extnID.
 With this understanding, the needs of {{RFC8994}} and {{RFC8995}} are satisfied
 with no change to the bits on the wire.
 
+## Alternative: Use of CSR templates
+
+{{RFC8295, Appendix B}} suggests an alternative for the piecemeal specification of attributes that {{RFC7030}} documented.
+Instead, a CSR object is returned with various fields filled out, and other fields waiting to be filled in.
+This is also the method defined CMP Update {{[I-D.ietf-lamps-cmp-updates}} and the Lightweight CMP profile {{I-D.ietf-lamps-lightweight-cmp-profile, Section 4.3.3}}, which uses a CSR template in CRMF.
+
+~~~~
+    CertificationRequestInfo ::= SEQUENCE {
+        version       INTEGER { v1(0) } (v1,...),
+        subject       Name,
+        subjectPKInfo SubjectPublicKeyInfo{{ PKInfoAlgorithms }},
+        attributes    [0] Attributes{{ CRIAttributes }}
+        }
+~~~~
+
+The entire CSR is included as a new Attribute valid with OID TBD.
```

Comment:
```suggestion
The CertificationRequestInfo is included as a new Attribute valid with OID TBD.  This avoids a useless fake signature in the CSR object.
The version code is always v1 (0). As shown in the example below, any empty values in the subject DN, and in any included X509v3 extensions are expected to be filled in by the client.

The SubjectPublicKeyInfo field must be present, but it MUST have an empty bit string for the key, as the server does not know what key will be used.  The server MAY specify (in the OID), the type of the key to use, but otherwise the OID type MUST be NULL.
```